src/audio/module_adapter/module/waves.c

```diff
@@ -232,31 +232,31 @@ static int waves_effect_check(struct comp_dev *dev)
 	/* todo use fallback to comp_verify_params when ready */
 
 	/* resampling not supported */
-	if (src_fmt->rate != audio_stream_get_rate(snk_fmt)) {
+	if (audio_stream_get_rate(src_fmt) != audio_stream_get_rate(snk_fmt)) {
```

Comment:
oh, yeah, that's the punishment for being too inventive with names, we couldn't possibly have guessed to grep for `src_fmt` as a buffer stream name ;-)